### ISSUE TITLE
Router class name must be prefixed with $

### ISSUE THREAD

**Issue by tyypgzl** [Milad-Akarie/auto_route_library#1175]
First time using auto route but runner is not working to me.

<details>
<summary>pubspec.yaml</summary>
<br>

dependencies:
  flutter:
    sdk: flutter

  cupertino_icons: ^1.0.2
  auto_route: ^4.2.1

dev_dependencies:
  flutter_test:
    sdk: flutter

  flutter_lints: ^2.0.0
  auto_route_generator: ^4.2.0
  build_runner: ^2.2.0
</details>

<details>
<summary>flutter doctor</summary>
<br>

[✓] Flutter (Channel stable, 3.0.5, on macOS 12.4 21F79 darwin-arm, locale tr-TR)
    • Flutter version 3.0.5 at /Users/tayyipguzel/Desktop/development/src/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f1875d570e (3 hafta önce), 2022-07-13 11:24:16 -0700
    • Engine revision e85ea0e79c
    • Dart version 2.17.6
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/tayyipguzel/Library/Android/sdk
    • Platform android-31, build-tools 33.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] VS Code (version 1.69.2)
    • VS Code at /Users/tayyipguzel/Downloads/Visual Studio Code.app/Contents
    • Flutter extension version 3.46.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 12.4 21F79 darwin-arm
    • Chrome (web)    • chrome • web-javascript • Google Chrome 103.0.5060.134

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
</details>

```
import 'package:auto_flutter_route/home_detail_view.dart';
import 'package:auto_flutter_route/home_view.dart';
import 'package:auto_route/auto_route.dart';

part 'app_router.gr.dart';

@MaterialAutoRouter(
  replaceInRouteName: 'View,Route',
  routes: <AutoRoute>[
    AutoRoute(page: HomeView, initial: true, path: '/home'),
    AutoRoute(page: HomeDetailView, path: '/homedetail'),
  ],
)
class AppRouter extends _$AppRouter {}
```

ı create auto_router.dart file and copy the code above. Then I run the build runner command. 
```
flutter packages pub run build_runner build
```

```
[INFO] Generating build script...
[INFO] Generating build script completed, took 160ms

[INFO] Initializing inputs
[INFO] Reading cached asset graph...
[INFO] Reading cached asset graph completed, took 29ms

[INFO] Checking for updates since last build...
[INFO] Checking for updates since last build completed, took 413ms

[INFO] Running build...
[INFO] Running build completed, took 6ms

[INFO] Caching finalized dependency graph...
[INFO] Caching finalized dependency graph completed, took 17ms

[SEVERE] auto_route_generator:autoRouteGenerator on lib/navigator/auto_router.dart (cached):

Router class name must be prefixed with $
package:auto_flutter_route/navigator/auto_router.dart:14:7
   ╷
14 │ class AppRouter extends _$AppRouter {}
   │       ^^^^^^^^^
   ╵
[SEVERE] Failed after 30ms
pub finished with exit code 1
```
I got the above error.

**Comment by murataksuu** [comment]
can you try this?
`class AppRouter extends _$AppRouter {} -> class $AppRouter {}`

**Comment by tyypgzl** [comment]
> 
@murataksuu
I got the below error. 
Expected a method, getter, setter or operator declaration.
This appears to be incomplete code. Try removing it or completing it.

**Comment by LogicNg** [comment]
same problem here

**Comment by Milad-Akarie** [comment]
fixed in auto_route_generator 4.2.1
